test/test.jsx

```diff
@@ -1406,4 +1406,114 @@ test('should crash on a plugin replacing `root`', () => {
   }, /Expected a `root` node/)
 })
 
+/**
+ * @param {string} string
+ * @param {number} start
+ * @param {number} end
+ * @returns {import('unist').Literal<string>}
+ */
+const extractText = (string, start, end) => {
+  const startLine = string.slice(0, start).split('\n')
+  const endLine = string.slice(0, end).split('\n')
+  return {
+    type: 'text',
+    value: string.slice(start, end),
+    position: {
+      start: {
+        line: startLine.length,
+        column: startLine[startLine.length - 1].length + 1
+      },
+      end: {
+        line: endLine.length,
+        column: endLine[endLine.length - 1].length + 1
+      }
+    }
+  }
+}
+
+/** @type {import('unified').Plugin<Array<void>, import('unist').Literal<string>>} */
+const pluginCustomComponentType = () => {
+  const dRegex = /d/
+  /** @type {import('unified').Transformer<import('unist').Literal<string>>} */
+  const transformer = (tree) => {
+    visit(
+      /** @type {import('unist').Literal<string>} */ tree,
+      /** {@type string} */ 'text',
+      (
+        /** @type {import('unist').Literal<string>} */ node,
+        position,
+        /** @type {import('unist').Parent<import('unist').Literal<string>> | null} */ parent
+      ) => {
+        if (typeof node.value !== 'string') {
+          return
+        }
+
+        position = position ?? 0
+        /** @type {Array<import('unist').Literal<string>>} */
+        const definition = []
+        let lastIndex = 0
+        const match = dRegex.exec(node.value)
+        if (match !== null) {
+          const value = match[0]
+          if (match.index !== lastIndex) {
+            definition.push(extractText(node.value, lastIndex, match.index))
+          }
+
+          definition.push({type: 'customType', value})
```

Comment:
why not leverage a custom element type? (which should be supported today, without any changes)
```suggestion
          definition.push({type: 'element', tagName: 'custom-type', properties: { value }})
```

Reply:
It seems like, it is not working, by doing this I've got this browser error message:
```js
ast-to-react.js?002a:121 Uncaught TypeError: Cannot read properties of undefined (reading 'length')
    at childrenToReact (ast-to-react.js?002a:121)
    at toReact (ast-to-react.js?002a:185)
    at childrenToReact (ast-to-react.js?002a:125)
    at toReact (ast-to-react.js?002a:185)
    at childrenToReact (ast-to-react.js?002a:125)
    at ReactMarkdown (react-markdown.js?2a9f:114)
```

In case you are asking, I cleared the cache, uninstalled/reinstalled `node_modules`, it is using the latest version of `react-markdown`).

Indeed, if we could do something like that, it would solve the problem. :+1:

Reply:
https://codesandbox.io/s/custom-element-example-r4mt5

Reply:
Thank you very much, it works!
I will close the PR as no changes are needed.

I could open an issue before opening a PR, my mistake.